provider/adapters/src/fork/by_error.rs

```diff
@@ -142,10 +142,20 @@ impl<P, F> MultiForkByErrorProvider<P, F> {
         &self.providers
     }
 
+    /// Returns a mutable reference to the inner vector of providers.
+    pub fn inner_mut(&mut self) -> &mut Vec<P> {
```

Comment:
This method gives the caller the full range of `Vec` mutation functions, but it freezes the fact that the inner field is a `Vec` in the API (preventing us from swapping it for some other storage solution); thoughts?

Reply:
I think it's ... fine for now. if we need more we can give them separate names.